NTupleAnalyzerHH_nanoAOD.py

```diff
@@ -1094,132 +1158,281 @@ def LeptonsAndJetsFromHH(T):
+					sub2 = emptyvector
+				else:
+					sub2 = TLorentzVector()
+					sub2.SetPtEtaPhiM(SubJetPt[SubJetInd2[j]],SubJetEta[SubJetInd2[j]],SubJetPhi[SubJetInd2[j]],SubJetMass[SubJetInd2[j]])
+
+				subjet1.append(sub1)
+				subjet2.append(sub2)
+
+				mindr = 99999
+
+				for g in range(len(genjetsH)):
+
+					genjet = genjetsH[g]
+
+					dr1 = sub1.DeltaR(genjet) if sub1 else 99999
+					dr2 = sub2.DeltaR(genjet) if sub2 else 99999
+
+					mindrtotal = dr1 + dr2
+
+					if mindrtotal < mindr:
+						mindr1 = dr1
+						mindr2 = dr2
+						mindr = mindrtotal
+						bestindices = [j, g]
+
+			bestrecojetind = bestindices[0]
+
+			CSV_jetH = 0
+
+			if mindr1<0.6 and mindr2<0.6:
+				matchedrecojetH.append(jetsH[bestrecojetind])
 				recojetindsH.append(recojetOriIndsH[bestrecojetind])
 				usedHjetsInd.append(bestrecojetind)
-				#recojetindsH.append(bestrecojetind)
+				CSV_jetH = PFJetCombinedInclusiveSecondaryVertexBTag[bestrecojetind]
 			else:
-				matchedrecojetsH.append(emptyvector)
+				matchedrecojetH.append(emptyvector)
 				recojetindsH.append(-99)
-			#if v == '':print 'Done this genjetsH with bestrecojetind', bestrecojetind , 'mindr:',mindr, 'len(usedHjetsInd)', len(usedHjetsInd)
-		while len(recojetindsH)<2 :
-			recojetindsH.append(-99)
 
-		usedZjetsInd = []
-		for g in genjetsZ:
+			while len(recojetindsH)<1 :
+				recojetindsH.append(-99)
+
+###################### AK8 matching with 2 GEN GETS from Z ######################
+			usedZjetsInd = []
+			mindr1 = 99999
+			mindr2 = 99999
 			bestrecojetind=-1
-			mindr = 99999
-			ind=-1
-			for m in jetsZ:
-				ind+=1
+
+			for j in range(len(jetsZ)):
+
 				skip = False
-				for indc in range(len(usedZjetsInd)):
-					#if v == '':print 'indc', indc, 'jetind stored in usedZjetsInd', usedZjetsInd[indc]
-					if ind == usedZjetsInd[indc]:
+				for indc in range(len(usedHjetsInd)):
+					if j == usedHjetsInd[indc]:
 						skip = True
-						#if v == '':print ' >> skip ind in jetsZ', ind
 						continue
-				for indc in range(len(usedHjetsInd)):
-					if ind == usedHjetsInd[indc]:
+				for indc in range(len(usedZjetsInd)):
+					if j == usedZjetsInd[indc]:
 						skip = True
 						continue
 				if skip: continue
-				dr = abs(m.DeltaR(g))
-				#if v == '':print 'ind in jetsZ', ind ,'DeltaR:',dr, ' >> gen pt eta', g.Pt(),g.Eta(),' >> reco pt eta', m.Pt(),m.Eta(), 'dr', dr
-				if dr<mindr:
-					mindr =dr
-					bestrecojetind=ind
-			if mindr<0.6:
-				matchedrecojetsZ.append(jetsZ[bestrecojetind])
+
+				subjet1,subjet2 = [],[]
+
+				if SubJetInd1[j] == -1:
+					sub1 = emptyvector
+				else:
+					sub1 = TLorentzVector()
+					sub1.SetPtEtaPhiM(SubJetPt[SubJetInd1[j]],SubJetEta[SubJetInd1[j]],SubJetPhi[SubJetInd1[j]],SubJetMass[SubJetInd1[j]])
+
+				if SubJetInd2[j] == -1:
+					sub2 = emptyvector
+				else:
+					sub2 = TLorentzVector()
+					sub2.SetPtEtaPhiM(SubJetPt[SubJetInd2[j]],SubJetEta[SubJetInd2[j]],SubJetPhi[SubJetInd2[j]],SubJetMass[SubJetInd2[j]])
```

Comment:
This looks like a problem in the indentation compared to the old version, please check.